iothub/service/src/Common/Data/SharedAccessSignatureAuthorizationRule.cs

```diff
@@ -54,5 +54,37 @@ public bool Equals(SharedAccessSignatureAuthorizationRule other)
 
             return equals;
         }
+
+        public override bool Equals(object obj)
+        {
+            return Equals(obj as SharedAccessSignatureAuthorizationRule);
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage(
```

Comment:
Rather than suppress the whole thing, what about doing it conditionally based on the framework?

Reply:
let me give that a shot and see how it would look like. I wanted to avoid adding more conditional logic based on framework but I guess it's better than supressing the whole method.

Reply:
done